pyonfx/utils.py

```diff
@@ -33,8 +33,8 @@ class Utils:
 
     @staticmethod
     def all_non_empty(
-        lines_chars_syls_or_words: List[Union[Line, Word, Syllable, Char]]
-    ) -> List[Union[Line, Word, Syllable, Char]]:
+        lines_words_syls_or_chars: Union[List[Line], List[Word], List[Syllable], List[Char]],
```

Comment:
Isn't that a more complicate way to write what I wrote on the left?
Same for line 37 below.

Reply:
linters complaint about it even by just passing a basic List[Char] because of convariance stuff.
See https://mypy.readthedocs.io/en/stable/common_issues.html?highlight=covariance#invariance-vs-covariance.

Actually I think we should move this function as class method in the respective classes. Types should be more easy to handle.

Reply:
The return type is not really better than the old one since the result is the same when you want to use in a loop.

Reply:
Oh it's actually related to #28

Reply:
Commit https://github.com/CoffeeStraw/PyonFX/pull/33/commits/1bceb5bd6105edf5417fdca6b21c02482f36c8b2 should definitely fix the return type